test/unit/JobsManager.js

```diff
@@ -40,176 +47,288 @@ contract("JobsManager", accounts => {
+            assert.equal(jInfo[5], currentRound, "wrong creationRound")
+            assert.equal(jInfo[6], currentBlock, "wrong creationBlock")
+            assert.equal(jInfo[7], currentBlock + 50, "wrong endBlock")
+            assert.equal(jInfo[8], 0, "wrong escrow")
+            assert.equal(jInfo[9], 0, "wrong totalClaims")
+        })
+
+        it("should create a new NewJob event", async () => {
             const e = jobsManager.NewJob({})
 
             e.watch(async (err, result) => {
                 e.stopWatching()
 
-                assert.equal(result.args.broadcaster, broadcaster, "broadcaster incorrect")
-                assert.equal(result.args.jobId, 0, "new job id incorrect")
-                assert.equal(result.args.streamId, streamId, "stream id incorrect")
-                assert.equal(result.args.transcodingOptions, transcodingOptions, "transcoding options incorrect")
+                assert.equal(result.args.broadcaster, broadcaster, "wrong broadcaster")
+                assert.equal(result.args.jobId, 0, "wrong jobId")
+                assert.equal(result.args.streamId, "foo", "wrong streamId")
+                assert.equal(result.args.transcodingOptions, transcodingOptions, "wrong transcodingOptions")
             })
 
-            const endBlock = 100 + 500
-            await jobsManager.job(streamId, transcodingOptions, maxPricePerSegment, endBlock, {from: broadcaster})
+            await jobsManager.job("foo", transcodingOptions, 1, currentBlock + 50, {from: broadcaster})
         })
 
-        it("should create a new job", async () => {
-            const creationRound = 100
-            await fixture.roundsManager.setCurrentRound(creationRound)
+        it("should set the broadcaster's withdraw block to the job's end block if the job's end block > broadcaster's withdraw block", async () => {
+            await jobsManager.deposit({from: broadcaster, value: 1000})
+            await jobsManager.job("foo", transcodingOptions, 1, currentBlock + 50, {from: broadcaster})
 
-            const transcoderTotalStake = 100
-            await fixture.bondingManager.setActiveTranscoder(accounts[1], 0, transcoderTotalStake, 0)
+            const bInfo = await jobsManager.broadcasters.call(broadcaster)
+            assert.equal(bInfo[1], currentBlock + 50, "wrong broadcaster withdrawBlock")
+        })
 
-            const endBlock = 100 + 500
-            const creationBlock = 100
-            await jobsManager.job(streamId, transcodingOptions, maxPricePerSegment, endBlock, {from: broadcaster})
+        it("should not change the broadcaster's withdraw block if the job's end block <= broadcaster's withdraw block", async () => {
+            await jobsManager.deposit({from: broadcaster, value: 1000})
+            await jobsManager.job("foo", transcodingOptions, 1, currentBlock + 50, {from: broadcaster})
+            await jobsManager.job("foo", transcodingOptions, 1, currentBlock + 50, {from: broadcaster})
```

Comment:
Yeah since the second job is created with the same end block as the first job, this is testing that the creation of a second job with an end block <= the broadcaster's withdraw block leaves the broadcaster's withdraw block unchanged.